### PR TITLE
Task manager home

### PR DESCRIPTION
**Pull request by vingtle**

### CODE REVIEW

**Review by Erwan60 — APPROVED** [review]

File: client/src/components/TaskManager/TaskManger.jsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/* eslint-disable no-unused-vars */
```
Comment (Brian-boudrioux) [inline client/src/components/TaskManager/TaskManger.jsx]: remove eslint-disable lines here

File: client/src/components/TaskManager/TaskManger.jsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+import "./taskmanager.css";
+import { IoIosAddCircleOutline } from "react-icons/io";
+import { RiDeleteBin5Line } from "react-icons/ri";
+
+function TaskManager() {
+  const [currentTab, setCurrentTab] = useState("todo");
+  const [newTaskText, setNewTaskText] = useState("");
+  const [isInputVisible, setInputVisible] = useState(false);
+  const [buttonVisible, setButtonVisible] = useState(true);
+  const [tasks, setTasks] = useState({
```
Comment (Brian-boudrioux) [inline client/src/components/TaskManager/TaskManger.jsx]: Fetch the tasks from the api inside a useEffect to update your state

File: client/src/components/TaskManager/TaskManger.jsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    taskListRef.current.scrollTop = 0;
+  };
+
+  function toggleInputVisible() {
+    setInputVisible(!isInputVisible);
+    setButtonVisible(!buttonVisible);
+  }
+
+  function addTask() {
+    if (!newTaskText.trim()) return;
```
Comment (Brian-boudrioux) [inline client/src/components/TaskManager/TaskManger.jsx]: do not touch create and delete tasks function for the moment, next features

File: server/app/controllers/taskActions.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        const [tasks] = await tables.task.getAll();
+        res.status(200).json(tasks);
+    } catch (err) {
+        next(err);
+    }
+};
+
+const getByStatus = async (req, res, next) => {
+    try {
+        const status = req.body;
```
Comment (Brian-boudrioux) [inline server/app/controllers/taskActions.js]: use destructuration  to get the status from the params object : const { status } = req.params

**Review by Brian-boudrioux — CHANGES_REQUESTED** [review]

File: client/src/components/TaskManager/TaskManger.jsx (hunk 1 above)
Comment (vingtle) [reply client/src/components/TaskManager/TaskManger.jsx]: done

File: client/src/components/TaskManager/TaskManger.jsx (hunk 2 above)
Comment (vingtle) [reply client/src/components/TaskManager/TaskManger.jsx]: done

File: client/src/components/TaskManager/TaskManger.jsx (hunk 3 above)
Comment (vingtle) [reply client/src/components/TaskManager/TaskManger.jsx]: done

File: server/app/controllers/taskActions.js (hunk 4 above)
Comment (vingtle) [reply server/app/controllers/taskActions.js]: done

Comment (vingtle) [reply server/app/controllers/taskActions.js]: done

File: client/src/components/TaskManager/TaskManger.jsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+import "./taskmanager.css";
+import { IoIosAddCircleOutline } from "react-icons/io";
+import { RiDeleteBin5Line } from "react-icons/ri";
+
+function TaskManager() {
+  const [currentTab, setCurrentTab] = useState("todo");
+  const [newTaskText, setNewTaskText] = useState("");
+  const [isInputVisible, setInputVisible] = useState(false);
+  const [buttonVisible, setButtonVisible] = useState(true);
+  const [tasks, setTasks] = useState([]);
```
Comment (Alexandre78R) [inline client/src/components/TaskManager/TaskManger.jsx]: replace [] --> {}

File: client/src/pages/Connection/Connection.jsx [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
-        <h1>connexion</h1>
-        <p>Bienvenue sur la page d'accueil.</p>
-      </div>
-    );
-  }
-  
-  export default Connection;
\ No newline at end of file
+  return (
+    <div>
```
Comment (Alexandre78R) [inline client/src/pages/Connection/Connection.jsx]: replace fragement react please

File: client/src/pages/Member/Member.jsx [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
-    return (
-      <div>
-        <h1>Member</h1>
-      </div>
-    );
-  }
-  
-  export default Member;
+  return (
+    <div>
```
Comment (Alexandre78R) [inline client/src/pages/Member/Member.jsx]: replace fragement react

File: client/src/components/TaskManager/TaskManger.jsx
Comment (Alexandre78R) [inline client/src/components/TaskManager/TaskManger.jsx:1]: Rename TaskManger --> TaskManager

File: client/src/components/TaskManager/taskmanager.css [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+.span {
+  display: flex;
+  font-size: 20px;
+  color: #6499ff;
+  margin-top: 10px;
+  margin-bottom: 10px;
+  padding-left: 45%;
+}
+.tabs {
+  display: felx;
```
Comment (Alexandre78R) [inline client/src/components/TaskManager/taskmanager.css]: rename flex

File: client/src/pages/Archive/Archive.jsx [hunk 9]
```diff
@@ -1,14 +1,11 @@
...
-        <h1>Archive</h1>
-       
-      </div>
-    );
-  }
-  
-  export default Archive ;
\ No newline at end of file
+  return (
+    <div>
```
Comment (Alexandre78R) [inline client/src/pages/Archive/Archive.jsx]: replace fragement react

File: client/src/pages/Welcome/welcome.css [hunk 10]
```diff
@@ -114,3 +116,90 @@ text {
...
+}
+
+@media (max-width: 480px) {
+  .login-btn,
+  .signup-btn {
+    padding: 6px 15px;
+    font-size: 14px;
+  }
+
+  .logo-devlog {
```
Comment (Alexandre78R) [inline client/src/pages/Welcome/welcome.css]: delete doublon and triplon width: 150px

File: client/src/pages/Welcome/welcome.css [hunk 11]
```diff
@@ -114,3 +116,90 @@ text {
...
+  .signup-btn {
+    padding: 12px 25px;
+    font-size: 18px;
+  }
+  .group-button {
+    top: 15px;
+    right: 15px;
+  }
+  .logo-devlog {
+    width: 150px;
```
Comment (Alexandre78R) [inline client/src/pages/Welcome/welcome.css:183]: delete width 150px...

File: server/database/schema.sql [hunk 12]
```diff
@@ -16,31 +37,50 @@ USE `devlog` ;
 -- -----------------------------------------------------
 DROP TABLE IF EXISTS `devlog`.`User` ;
 
+>>>>> dev
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 13]
```diff
@@ -16,31 +37,50 @@ USE `devlog` ;
...
 CREATE TABLE IF NOT EXISTS `devlog`.`User` (
   `id` INT NOT NULL AUTO_INCREMENT,
   `username` VARCHAR(50) NOT NULL,
   `email` VARCHAR(100) NOT NULL,
   `password` VARCHAR(255) NOT NULL,
   `is_admin` TINYINT NOT NULL DEFAULT 0,
   PRIMARY KEY (`id`),
+<<<<<TaskManager-Home
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 14]
```diff
@@ -16,31 +37,50 @@ USE `devlog` ;
...
+  UNIQUE INDEX `id_UNIQUE` (`id` ASC) VISIBLE
+) ENGINE = InnoDB;
+-- -----------------------------------------------------
+-- Table `devlog`.`Team`
+-- -----------------------------------------------------
+DROP TABLE IF EXISTS `devlog`.`Team`;
+=======
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 15]
```diff
@@ -16,31 +37,50 @@ USE `devlog` ;
...
   UNIQUE INDEX `id_UNIQUE` (`id` ASC) VISIBLE)
 ENGINE = InnoDB;
 -- -----------------------------------------------------
 -- Table `devlog`.`Team`
 -- -----------------------------------------------------
 DROP TABLE IF EXISTS `devlog`.`Team` ;
 
+>>>>>> dev
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 16]
```diff
@@ -16,31 +37,50 @@ USE `devlog` ;
...
 CREATE TABLE IF NOT EXISTS `devlog`.`Team` (
   `id` INT NOT NULL AUTO_INCREMENT,
   `name` VARCHAR(55) NOT NULL,
   PRIMARY KEY (`id`),
+<<<< TaskManager-Home
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 17]
```diff
@@ -16,31 +37,50 @@ USE `devlog` ;
...
+DROP TABLE IF EXISTS `devlog`.`Project`;
+
   UNIQUE INDEX `id_UNIQUE` (`id` ASC) VISIBLE)
 ENGINE = InnoDB;
 -- -----------------------------------------------------
 -- Table `devlog`.`Project`
 -- -----------------------------------------------------
 DROP TABLE IF EXISTS `devlog`.`Project` ;
 
+>>>>> dev
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 18]
```diff
@@ -60,15 +108,28 @@ ENGINE = InnoDB;
 -- -----------------------------------------------------
 DROP TABLE IF EXISTS `devlog`.`Task` ;
 
+>>>>> dev
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 19]
```diff
@@ -60,15 +108,28 @@ ENGINE = InnoDB;
...
 CREATE TABLE IF NOT EXISTS `devlog`.`Task` (
   `id` INT NOT NULL AUTO_INCREMENT,
   `name` VARCHAR(50) NOT NULL,
   `description` VARCHAR(250) NOT NULL,
+<<<<< TaskManager-Home
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 20]
```diff
@@ -60,15 +108,28 @@ ENGINE = InnoDB;
...
+  `is_archived` TINYINT NOT NULL,
+  
   `is_achived` TINYINT NOT NULL,
+ dev
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 21]
```diff
@@ -60,15 +108,28 @@ ENGINE = InnoDB;
...
   `Project_id1` INT NOT NULL,
   PRIMARY KEY (`id`),
   INDEX `fk_Task_Project1_idx` (`Project_id1` ASC) VISIBLE,
   UNIQUE INDEX `id_UNIQUE` (`id` ASC) VISIBLE,
+<<<<<TaskManager-Home
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 22]
```diff
@@ -81,6 +142,7 @@ ENGINE = InnoDB;
 -- -----------------------------------------------------
 DROP TABLE IF EXISTS `devlog`.`User_has_Team` ;
 
+>>>>> dev
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 23]
```diff
@@ -90,6 +152,15 @@ CREATE TABLE IF NOT EXISTS `devlog`.`User_has_Team` (
   INDEX `fk_User_has_Team_User1_idx` (`User_id` ASC) VISIBLE,
   UNIQUE INDEX `User_id_UNIQUE` (`User_id` ASC) VISIBLE,
   UNIQUE INDEX `Team_id_UNIQUE` (`Team_id` ASC) VISIBLE,
+<<<<< TaskManager-Home
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 24]
```diff
@@ -90,6 +152,15 @@ CREATE TABLE IF NOT EXISTS `devlog`.`User_has_Team` (
...
+  CONSTRAINT `fk_User_has_Team_User1` FOREIGN KEY (`User_id`) REFERENCES `devlog`.`User` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION,
+  CONSTRAINT `fk_User_has_Team_Team1` FOREIGN KEY (`Team_id`) REFERENCES `devlog`.`Team` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION
+) ENGINE = InnoDB;
+-- -----------------------------------------------------
+-- Table `devlog`.`User_has_Task`
+-- -----------------------------------------------------
+DROP TABLE IF EXISTS `devlog`.`User_has_Task`;
+=======
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 25]
```diff
@@ -107,6 +178,7 @@ ENGINE = InnoDB;
 -- -----------------------------------------------------
 DROP TABLE IF EXISTS `devlog`.`User_has_Task` ;
 
+>>>>>> dev
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 26]
```diff
@@ -116,6 +188,14 @@ CREATE TABLE IF NOT EXISTS `devlog`.`User_has_Task` (
   INDEX `fk_User_has_Task_User1_idx` (`User_id` ASC) VISIBLE,
   UNIQUE INDEX `User_id_UNIQUE` (`User_id` ASC) VISIBLE,
   UNIQUE INDEX `Task_id_UNIQUE` (`Task_id` ASC) VISIBLE,
+<<<<<<TaskManager-Home
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql [hunk 27]
```diff
@@ -130,4 +210,5 @@ ENGINE = InnoDB;
 
 SET SQL_MODE=@OLD_SQL_MODE;
 SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS;
-SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS;
\ No newline at end of file
+SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS;
+>>>>>> dev
```
Comment (Alexandre78R) [inline server/database/schema.sql]: delete conflit

File: server/database/schema.sql
Comment (Alexandre78R) [inline server/database/schema.sql:1]: verify all file conflit

File: server/app/routers/api/router.js
Comment (Alexandre78R) [inline server/app/routers/api/router.js:1]: No doublon route ! Warning  : conflit !

**Review by Alexandre78R — CHANGES_REQUESTED** [review]

File: client/src/components/TaskManager/TaskManger.jsx (hunk 5 above)
Comment (vingtle) [reply client/src/components/TaskManager/TaskManger.jsx]: done

File: client/src/pages/Connection/Connection.jsx (hunk 6 above)
Comment (vingtle) [reply client/src/pages/Connection/Connection.jsx]: done

File: server/app/routers/api/router.js
Comment (Alexandre78R) [inline server/app/routers/api/router.js:1]: No duplicate routes ! delete doublon route

File: client/src/pages/Member/Member.jsx (hunk 7 above)
Comment (vingtle) [reply client/src/pages/Member/Member.jsx]: done

File: client/src/components/TaskManager/TaskManger.jsx
Comment (vingtle) [reply client/src/components/TaskManager/TaskManger.jsx:1]: done

File: client/src/components/TaskManager/taskmanager.css (hunk 8 above)
Comment (vingtle) [reply client/src/components/TaskManager/taskmanager.css]: done

File: server/database/schema.sql (hunk 25 above)
Comment (vingtle) [reply server/database/schema.sql]: done

File: server/database/schema.sql (hunk 26 above)
Comment (vingtle) [reply server/database/schema.sql]: done

File: server/database/schema.sql (hunk 27 above)
Comment (vingtle) [reply server/database/schema.sql]: done

File: server/database/schema.sql
Comment (vingtle) [reply server/database/schema.sql:1]: done

File: server/app/routers/api/router.js
Comment (vingtle) [reply server/app/routers/api/router.js:1]: done

File: client/src/components/TaskManager/TaskManager.jsx [hunk 28]
```diff
@@ -0,0 +1,160 @@
...
+    setTasks((prevTasks) => ({
+      ...prevTasks,
+      todo: [...prevTasks.todo, newTask],
+    }));
+    setNewTaskText("");
+    setInputVisible(false);
+    setButtonVisible(true);
+  }
+
+  function deleteTask(taskId) {
```
Comment (Alexandre78R) [inline client/src/components/TaskManager/TaskManager.jsx:66]: add fetch delete task backend

File: client/src/components/TaskManager/TaskManager.jsx [hunk 29]
```diff
@@ -0,0 +1,160 @@
...
+    taskListRef.current.scrollTop = 0;
+    }
+  };
+
+  function toggleInputVisible() {
+    setInputVisible(!isInputVisible);
+    setButtonVisible(!buttonVisible);
+  }
+
+  function addTask() {
```
Comment (Alexandre78R) [inline client/src/components/TaskManager/TaskManager.jsx:50]: add fetch addTask backend

File: server/app/controllers/taskActions.js [hunk 30]
```diff
@@ -0,0 +1,27 @@
...
+        if (!userId || !status) {
+            return res.status(400).json({ message: 'Missing required information: userId and status' });
+        }
+        const [results] = await tables.task.create({ text, status, userId });
+        res.status(201).json(results);
+    } catch (error) {
+        next(error);
+    } return true;
+};
+
```
Comment (Alexandre78R) [inline server/app/controllers/taskActions.js:26]: create controller delete Task

File: server/app/routers/api/tasks/router.js [hunk 31]
```diff
@@ -0,0 +1,11 @@
+const express = require("express");
+
+const router = express.Router();
+
+const { browse, addTask } = require("../../../controllers/taskActions");
+
+router.get("/", browse);
+
+router.post("/add", addTask);
+
```
Comment (Alexandre78R) [inline server/app/routers/api/tasks/router.js:10]: add router delete task

**Review by Alexandre78R — CHANGES_REQUESTED** [review]

File: server/app/routers/api/router.js
Comment (vingtle) [reply server/app/routers/api/router.js:1]: done

**Review by Alexandre78R — APPROVED** [review]
Good

**Pull request merged**